component/attr_cache/attr_cache.go

```diff
@@ -139,22 +138,18 @@ func (ac *AttrCache) Configure(_ bool) error {
 		ac.cacheTimeout = defaultAttrCacheTimeout
 	}
 
-	if config.IsSet(compName + ".cache-on-list") {
-		ac.cacheOnList = conf.CacheOnList
-	} else {
```

Comment:
So it's more like only if user has given the value we want to honor that otherwise use the default one. When we use structure we have no way to know whether user set the value or its coming from default one so we always check whether the flag was set explicitly by user or not.